crates/rpc/rpc/src/eth/helpers/transaction.rs

```diff
@@ -41,6 +45,48 @@ where
 
         Ok(hash)
     }
+
+    async fn send_raw_transaction_sync(&self, tx: Bytes) -> Result<Receipt, Self::Error> {
+        let recovered = recover_raw_transaction(&tx)?;
+
+        // broadcast raw transaction to subscribers if there is any.
+        self.broadcast_raw_transaction(tx);
+
+        let pool_transaction = <Self::Pool as TransactionPool>::Transaction::from_pooled(recovered);
+        // submit the transaction to the pool with a Local origin
+        let hash = self
+            .pool()
+            .add_transaction(TransactionOrigin::Local, pool_transaction)
+            .await
+            .map_err(Self::Error::from_eth_err)?;
+
+        let receiver = self.provider().subscribe_to_canonical_state();
+        let mut stream = self.provider().canonical_state_stream();
+        //let mut stream = tokio_stream::wrappers::BroadcastStream::new(receiver);
+        let mut timeout_sleep = tokio::time::sleep(tokio::time::Duration::from_secs(30));
+        tokio::pin!(timeout_sleep);
+
+        loop {
+            tokio::select! {
+                maybe_notification = stream.next() => {
+                    match maybe_notification {
+                        Some(notification) => {
+                            let chain = notification.committed();
+                            for block in chain.blocks_iter() {
+                                    // TODO: Receipt logic here
```

Comment:
for the receipt logic, we can utilize the existing

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/helpers/transaction.rs#L150-L150

once we know the tx was mined this should succeed

Reply:
then we also only need to check if any new block contains the hash, if so call transaction_receipt